graylog2-web-interface/src/components/indices/IndexSetFieldTypesList.tsx

```diff
@@ -0,0 +1,184 @@
+    sort: layoutConfig.sort,
+  }), [activePage, layoutConfig.pageSize, layoutConfig.sort, query]);
+  const { mutate: updateTableLayout } = useUpdateUserLayoutPreferences(ENTITY_TABLE_ID);
+  const onPageChange = useCallback(
+    (newPage: number, newPageSize: number) => {
+      if (newPage) {
+        setActivePage(newPage);
+      }
+
+      if (newPageSize) {
+        updateTableLayout({ perPage: newPageSize });
+      }
+    }, [updateTableLayout],
+  );
+
+  const onPageSizeChange = useCallback((newPageSize: number) => {
+    setActivePage(1);
+    updateTableLayout({ perPage: newPageSize });
+  }, [updateTableLayout]);
+
+  const onSortChange = useCallback((newSort: Sort) => {
+    setActivePage(1);
+    updateTableLayout({ sort: newSort });
+  }, [updateTableLayout]);
+
+  const onColumnsChange = useCallback((displayedAttributes: Array<string>) => {
+    updateTableLayout({ displayedAttributes });
+  }, [updateTableLayout]);
+  const { isLoading, data: { list, pagination, attributes }, refetch } = useIndexSetFieldTypes(indexSetId, searchParams, { enabled: !isLoadingLayoutPreferences });
+
+  const customColumnRenderers = useMemo(() => ({
+    attributes: {
+      type: {
+        renderCell: (item: string) => <span>{fieldTypes[item]}</span>,
+      },
+      is_custom: {
+        renderCell: (isCustom: boolean) => (isCustom ? <Icon title="Field has custom field type" name="check" /> : null),
+        staticWidth: 120,
+      },
+      is_reserved: {
+        renderCell: (isReserved: boolean) => (isReserved ? <Icon title="Field has reserved field type" name="check" /> : null),
+        staticWidth: 120,
+      },
+    },
+  }), [fieldTypes]);
+
+  const openEditModal = useCallback((fieldType: IndexSetFieldType) => {
+    handleOnOpen(fieldType);
+  }, [handleOnOpen]);
+
+  const renderActions = useCallback((fieldType: IndexSetFieldType) => {
+    const title = fieldType.isReserved ? 'Reserved field is not editable' : `Edit field type for ${fieldType.fieldName}`;
+
+    return (
+      <Button onClick={() => openEditModal(fieldType)}
+              role="button"
+              bsSize="xsmall"
+              disabled={fieldType.isReserved}
+              title={title}
+              tabIndex={0}>
+        Edit
+      </Button>
+    );
+  }, [openEditModal]);
+
+  if (isLoadingLayoutPreferences || isLoading) {
+    return <Spinner />;
+  }
+
+  return (
+    <>
+      <PaginatedList onChange={onPageChange}
+                     totalItems={pagination?.total}
+                     pageSize={layoutConfig.pageSize}
+                     activePage={activePage}
+                     showPageSizeSelect={false}
+                     useQueryParameter={false}>
+        {pagination?.total === 0 && !searchParams.query && (
+          <NoEntitiesExist>
+            No fields have been created yet.
+          </NoEntitiesExist>
+        )}
+        {!!list?.length && (
+          <EntityDataTable<IndexSetFieldType> data={list}
```

Comment:
Something seems to be broken with the state handling, when I have 40 Fields, set the page size to 10, go to page 2, the page size is 20 again and I only see one item.